app/submitter/converter.py

```diff
@@ -89,11 +89,17 @@ def convert_answers_to_data(answer_store, questionnaire_json, routing_path):
         answer_schema_list = SchemaHelper.get_answers_by_id_for_block(block_json)
 
         for answer in answers_in_block:
-            answer_schema = answer_schema_list[answer['answer_id']]
+            try:
+                answer_schema = answer_schema_list[answer['answer_id']]
+            except KeyError:
+                logger.error("No matching answer schema found in the store")
+                return data
```

Comment:
Thinking about it, this should be continue not return.
As if you have multiple answers it will return after the 1st one.
You should add a test with a mixture of valid and invalid answers and test that all of the value answers are returned by this method